src/BLAS/level_2_3/generator.jl

```diff
@@ -0,0 +1,110 @@
+# The increments to the `__counter` variable is for testing purposes
+
+function _check_transa(t::Char)
+    if !(t in ('C', 'N', 'T'))
+      error("transa: is '$t', must be 'N', 'T', or 'C'")
+    end
+end
+
+mkl_size(t::Char, M::AbstractVecOrMat) = t == 'N' ? size(M) : reverse(size(M))
+
+
+# Checks sizes for the multiplication C <- A * B
+function _check_mat_mult_matvec(C, A, B, tA)
+    _size(v::AbstractMatrix) = size(v)
+    _size(v::AbstractVector) = (size(v,1), 1)
+    _str(v::AbstractMatrix) = string("[", size(v,1), ", ", size(v,2), "]")
+    _str(v::AbstractVector) = string("[", size(v,1), "]")
+    mA, nA = mkl_size(tA, A)
+    mB, nB = _size(B)
+    mC, nC = _size(C)
+    if nA != mB || mC != mA || nC != nB
+        t = ""
+        if tA == 'T'; t = ".\'"; end
+        if tA == 'C'; t = "\'"; end
+        str = string("arrays had inconsistent dimensions for C <- A", t, " * B: ", _str(C), " <- ", _str(A), t, " * ", _str(B))
+        throw(DimensionMismatch(str))
+    end
+end
+
+for (mv, sv, mm, sm, T) in ((:mkl_scscmv, :mkl_scscsv, :mkl_scscmm, :mkl_scscsm, :Float32),
+                            (:mkl_dcscmv, :mkl_dcscsv, :mkl_dcscmm, :mkl_dcscsm, :Float64),
+                            (:mkl_ccscmv, :mkl_ccscsv, :mkl_ccscmm, :mkl_ccscsm, :Complex64),
+                            (:mkl_zcscmv, :mkl_zcscsv, :mkl_zcscmm, :mkl_zcscsm, :Complex128))
+@eval begin
+function cscmv!(transa::Char, α::$T, matdescra::String,
+                A::SparseMatrixCSC{$T, BlasInt}, x::StridedVector{$T},
+                β::$T, y::StridedVector{$T})
+    _check_transa(transa)
+    _check_mat_mult_matvec(y, A, x, transa)
+    global __counter[] += 1
```

Comment:
could be noticeable for small matrices. Pkg.test uses whatever you start it from, I think